go/cmd/dolt/commands/diff.go

```diff
@@ -886,27 +1225,28 @@ func diffRows(
 		}
 
 		// instantiate a new RowWriter with the new schema that only contains the columns with changes
-		rowWriter, err = dw.RowWriter(ctx, td, filteredUnionSch)
+		rowWriter, err = dw.RowWriter(fromTableInfo, toTableInfo, tableSummary, filteredUnionSch)
 		if err != nil {
 			return errhand.VerboseErrorFromError(err)
 		}
-		defer rowWriter.Close(ctx)
+		defer rowWriter.Close(sqlCtx)
 
 		// reset the row iterator
-		err = rowIter.Close(ctx)
+		err = rowIter.Close(sqlCtx)
 		if err != nil {
 			return errhand.BuildDError("Error closing row iterator:\n%s", query).AddCause(err).Build()
 		}
-		_, rowIter, err = sqlEng.Query(ctx, query)
-		defer rowIter.Close(ctx)
+		//_, rowIter, err = sqlEng.Query(ctx, query)
```

Comment:
delete dead line. Forget the past! On to the future!

Reply:
Removing.